package.json

```diff
@@ -16,8 +21,19 @@
 	"author": "Jacob Gunther",
 	"license": "MIT",
 	"devDependencies": {
-		"bluebird": "^3.7.2",
-		"jest": "^26.4.0"
+		"@types/chai": "^4.2.12",
+		"@types/chai-as-promised": "^7.1.3",
+		"@types/mocha": "^8.0.3",
+		"@types/node": "^12.12.54",
+		"@typescript-eslint/eslint-plugin": "^3.10.1",
+		"@typescript-eslint/parser": "^3.10.1",
+		"chai": "^4.2.0",
+		"chai-as-promised": "^7.1.1",
+		"eslint": "^7.7.0",
+		"mocha": "^8.1.1",
+		"rimraf": "^3.0.2",
+		"ts-node": "^9.0.0",
+		"typescript": "^4.0.2"
```

Comment:
Summary:

* All @types are purely so you can use these libraries in typescript.
* mocha is the test framework, mhai is for assertions. mhai-as-promised is for assertions on promises.
* ts-node is so that the tests can be written and run in typescript.
* eslint is still just eslint. You need the two additional plugins for it to work with typescript. These two are the replacement for tslint (now deprecated hence why its not used here).
* rimraf is to clean the build directory (`npm clean`)

Reply:
I understand what these dependencies are for, but how would one go about using this library within a normal Node.js application without TypeScript? Can it still be imported normally, and how is the support for older versions?

Reply:
The typescript is compiled to javascript based on the parameters in `tsconfig`. The typings are automatically generated.

This is a snapshot of what's exported right now (what the user has access to). Generated by the typescript compiler based on what you export in index.ts

```typescript
import { PingOptions, PingCallback, Response } from './model/api';
export declare function pingCallback(host: string, callback: PingCallback, options?: PingOptions): void;
export declare function ping(host: string, options?: PingOptions): Promise<Response>;
export * from './model/api'; 
```

Reply:
> I understand what these dependencies are for, but how would one go about using this library within a normal Node.js application without TypeScript? Can it still be imported normally, and how is the support for older versions?

So when you publish you only publish the contents of the `build` directory. Only the compiled javascript and typings are there.

Reply:
Ah, that makes a lot more sense now. I still may make my own TypeScript rewrite of the library to make the code more readable. I also have plans to implement a class for description formats for the next major rewrite (this one).

Reply:
Give this a review before you do that, most of the heavy duty work is done. You can just branch off of this and change what you'd like.